adapters/rubicon/rubicon_test.go

```diff
@@ -1306,14 +1320,410 @@ func TestOpenRTBRequestWithImpAndAdSlotIncluded(t *testing.T) {
 	if err := json.Unmarshal(rubiconReq.Imp[0].Ext, &rpImpExt); err != nil {
 		t.Fatal("Error unmarshalling imp.ext")
 	}
+	assert.Equal(t, rpImpExt.GPID, "/test-adslot")
+}
+
+func TestOpenRTBRequestUserParametersShouldBeBlocked(t *testing.T) {
+	SIZE_ID := getTestSizes()
+	bidder := new(RubiconAdapter)
+
+	request := &openrtb2.BidRequest{
+		ID: "test-request-id",
+		Imp: []openrtb2.Imp{{
+			ID: "test-imp-id",
+			Banner: &openrtb2.Banner{
+				Format: []openrtb2.Format{
+					SIZE_ID[15],
+					SIZE_ID[10],
+				},
+			},
+			Ext: json.RawMessage(`{
+				"bidder": {
+					"zoneId": 8394,
+					"siteId": 283282,
+					"accountId": 7891
+				}
+			}`),
+		}},
+		User: &openrtb2.User{
+			Yob: 200,
+			Geo: &openrtb2.Geo{
+				Lat:   123.46,
+				Lon:   678.89,
+				Metro: "some metro",
+				City:  "some city",
+				ZIP:   "some zip",
+			},
+			Gender:   "female",
+			Keywords: "someWords",
+		},
+		App: &openrtb2.App{
+			ID:   "com.test",
+			Name: "testApp",
+		},
+	}
+
+	reqs, _ := bidder.MakeRequests(request, &adapters.ExtraRequestInfo{})
```

Comment:
I'm curious why you opted to create several UTs that call `MakeRequests` when you could have created additional JSON tests. At a quick glance, I see that some of the fields in the original request that are incorporated into FPD like `keywords` and `inventory` don't appear in any of the JSON tests but are validated in some capacity in these UTs.

Reply:
To my mind, in future this test will be enriched with some scenarios, so decided to have it as UT

Reply:
Gotcha, the team chatted and would like to see the JSON framework utilized. In that case, could you please replace the tests you added that call `MakeRequests` with corresponding supplemental JSON tests?
That would be:
`TestOpenRTBRequestUserParametersShouldBeBlocked`
`TestOpenRTBRequestImpFpdFromAppResolving`
`TestOpenRTBRequestImpFpdFromSiteResolving`
`TestOpenRTBRequestUserFpdResolving`
`TestOpenRTBSiteSegTaxes` - I believe the test cases here can all be in a single JSON test

I understand that there are other tests in `rubicon_test.go` written a couple of years ago that take the UT approach with respect to `MakeRequests` but our docs now indicate that we require use of the JSON framework wherever possible as there are some advantages to using it.
https://docs.prebid.org/prebid-server/developers/add-new-bidder-go.html#adapter-code-tests